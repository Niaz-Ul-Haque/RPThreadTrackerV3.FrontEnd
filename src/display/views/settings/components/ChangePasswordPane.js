import React from 'react';
import { TabPane, Form, FormGroup, Col, Label, Row, Input, Button } from 'reactstrap';

const ChangePasswordPane = () => (
	<TabPane tabId="change-password">
		<Form action="" method="post" encType="multipart/form-data" className="form-horizontal">
			<FormGroup row>
				<Col xs="12" md={{ size: 3 }}>
					<Label htmlFor="current-password">Current Password:</Label>
				</Col>
				<Col xs="12" md="6">
					<Input
						type="password"
						id="current-password"
						name="current-password"
						placeholder="Enter Current Password"
					/>
				</Col>
			</FormGroup>
			<FormGroup row>
				<Col xs="12" md={{ size: 3 }}>
					<Label htmlFor="new-password">New Password:</Label>
				</Col>
				<Col xs="12" md="6">
					<Input
						type="password"
						id="new-password"
						name="new-password"
						placeholder="Enter New Password"
					/>
				</Col>
			</FormGroup>
			<FormGroup row>
				<Col xs="12" md={{ size: 3 }}>
					<Label htmlFor="confirm-new-password">Confirm New Password:</Label>
				</Col>
				<Col xs="12" md="6">
					<Input
						type="password"
						id="confirm-new-password"
						name="confirm-new-password"
						placeholder="Confirm New Password"
					/>
				</Col>
			</FormGroup>
		</Form>
		<Row>
			<Col sm="12" md={{ size: 8 }} className="text-right">
				<Button type="submit" color="primary">
					Submit
					</Button>
			</Col>
		</Row>
	</TabPane>
);

export default ChangePasswordPane;
