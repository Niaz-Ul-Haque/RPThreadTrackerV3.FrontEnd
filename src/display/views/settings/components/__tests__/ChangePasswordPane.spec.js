// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '~/utility/helpers.unit';
import ChangePasswordPane from '../ChangePasswordPane';
// #endregion imports

jest.mock('../../../../forms/TooltipForm', () => 'TooltipForm');
jest.mock('../../../../forms/change-password/ChangePasswordForm', () => 'ChangePasswordForm');

const createTestProps = propOverrides => ({
	submitChangePasswordForm: jest.fn(),
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const element = shallow(<ChangePasswordPane {...createTestProps()} />);
			expect(element).toMatchSnapshot();
		});
	});
});

describe('behavior', () => {
	describe('submitChangePasswordForm', () => {
		it('should be called when form is submitted', () => {
			const submitChangePasswordForm = jest.fn();
			const props = createTestProps({ submitChangePasswordForm });
			const element = shallow(<ChangePasswordPane {...props} />);
			const form = getSpecWrapper(element, 'change-password-form-container');
			form.props().onValidSubmit();
			expect(submitChangePasswordForm).toHaveBeenCalledTimes(1);
		});
	});
	describe('handleInputChange', () => {
		it('should handle text field update', () => {
			const event = { target: { type: 'text', name: 'password', value: 'my-new-password' } };
			const submitChangePasswordForm = jest.fn();
			const props = createTestProps({ submitChangePasswordForm });
			const element = shallow(<ChangePasswordPane {...props} />);
			element.instance().handleInputChange(event);
			element.update();
			const form = getSpecWrapper(element, 'change-password-form-container');
			form.props().onValidSubmit();
			expect(submitChangePasswordForm).toHaveBeenCalledTimes(1);
			expect(submitChangePasswordForm).toHaveBeenLastCalledWith({ password: 'my-new-password' });
		});
	});
});
