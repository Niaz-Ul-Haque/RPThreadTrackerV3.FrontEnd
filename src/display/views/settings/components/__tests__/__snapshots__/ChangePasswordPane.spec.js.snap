// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<TabPane
  tabId="change-password"
  tag="div"
>
  <Styled(Card)>
    <CardHeader
      tag="div"
    >
      <i
        className="fas fa-key"
      />
       Change Password
    </CardHeader>
    <CardBlock
      className="card-body"
    >
      <AvForm
        data-spec="change-password-form-container"
        method="get"
        model={Object {}}
        onInvalidSubmit={[Function]}
        onKeyDown={[Function]}
        onSubmit={[Function]}
        onValidSubmit={[Function]}
        tag={[Function]}
        validationEvent={
          Array [
            "onChange",
            "onInput",
          ]
        }
      >
        <TooltipForm
          Renderable="ChangePasswordForm"
          handleInputChange={[Function]}
        />
        <Row
          tag="div"
        >
          <Col
            className="text-right"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              color="primary"
              tag="button"
              type="submit"
            >
              Submit
            </Button>
          </Col>
        </Row>
      </AvForm>
    </CardBlock>
  </Styled(Card)>
</TabPane>
`;
